File: coding/18th February 2026/Binary Number with Alternating Bits.java
//Binary Number with Alternating Bits
class Solution 
{
    public boolean hasAlternatingBits(int n) 
    {
        StringBuilder str = new StringBuilder();
        while(n > 0)
        {
            int r = n % 2;
            n = n / 2;
            str.append(r);
        }
        for(int i = 1; i < str.length(); i++)
        {
            if((str.charAt(i - 1) == str.charAt(i)))
                return false;
        }
        return true;
    }
}

File: Coding in Python/18th February 2026/Binary Number with Alternating Bits.py
#Binary Number with Alternating Bits
class Solution:
    def hasAlternatingBits(self, n: int) -> bool:
        s = []
        while n > 0:
            r = n % 2
            n = n // 2
            s.append(r)
        for i in range(1, len(s)):
            if(s[i - 1] == s[i]):
                return False
        return True
